lfs_fuse_bd.c

```diff
@@ -93,16 +100,25 @@ int lfs_fuse_bd_prog(const struct lfs_config *cfg, lfs_block_t block,
     assert(block < cfg->block_count);
 
     // go to block
-    off_t err = lseek(fd, (off_t)block*cfg->block_size + (off_t)off, SEEK_SET);
+    off_t offrq = (off_t)block*cfg->block_size + (off_t)off;
+    off_t err = lseek(fd,offrq, SEEK_SET);
     if (err < 0) {
         return -errno;
+    } else if(err != offrq )
+    {
+        return LFS_ERR_IO;
     }
-
-    // write block
-    ssize_t res = write(fd, buffer, (size_t)size);
-    if (res < 0) {
-        return -errno;
-    }
+    const char* wrb = buffer;
+    do {
+        // write block
+        ssize_t res = write(fd, wrb, (size_t)size);
+        if (res <= 0) {
+            return (res==0)?(LFS_ERR_IO):(-errno);
+        } else {
+            wrb += res;
+            size -= res;
+        }
+    } while(0);
```

Comment:
I may be missing something, but why does this do/while loop have `0` as a condition?